certbot-apache/certbot_apache/configurator.py

```diff
@@ -2156,3 +2360,21 @@ def install_ssl_options_conf(self, options_ssl, options_ssl_digest):
         # to be modified.
         return common.install_version_controlled_file(options_ssl, options_ssl_digest,
             self.constant("MOD_SSL_CONF_SRC"), constants.ALL_SSL_OPTIONS_HASHES)
+
+    def generic_updates(self, _domain):
+        """
+        Interface method from interfaces.GenericUpdater to perform actions
+        when running certbot with "renew" verb, regardless of the lineage
+        being renewed.
+        """
+        self._autohsts_update()
```

Comment:
I was thinking of doing this later, as a part of a redesign of the complex enhancements but looks like it would make most sense to implement it as a part of this PR as well. I'm answering this in more detail in the comment about the interfaces, and old enhancement system.